### PR TITLE
Hide ToC from the Apache Airflow doc main page

### LINKED ISSUE #12554: Remove or limit table of content at the main Airflow doc page
Table of content here : http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/latest/index.html at the main page of "apache-airflow" documentation is huge and useless (especially that we have it in directory on the left). 

We should remove it or limit heavily (to 1st level only)

Screenshot from 2020-11-22 20-34-28

### PR DESCRIPTION
**Pull request by drdenzy**
Fixes #12554 - #12561 addressed this by limiting the depth, but we can do better.

### Before
before-changes-pr
before-changes-pr-2

### After
after-changes-pr

Hide the ToC from being displayed when the
the index.rst file in the Apache Airflow docs is rendered.

This will improve user experience and prevent repetition of what has
already been displayed on the sidebar.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by kaxil — APPROVED** [review]
For now this looks good, however I think it would be good to have a separate page for just "TOC/Index"

File: docs/index.rst [hunk 1]
```diff
@@ -69,10 +69,8 @@ similar from a run to the next, this allows for clarity around
 unit of work and continuity.
 
 
-Content
--------
 .. toctree::
-    :maxdepth: 1
+    :hidden:
```
Comment (kaxil) [inline docs/index.rst]:
```suggestion
    :hidden:
```

**Comment by ashb** [comment]
The whole point is the ToC is in the sidebar of every page.

**Comment by kaxil** [comment]
> The whole point is the ToC is in the sidebar of every page.

The problem with that is the sidebar hides nesting until you click on the level above it. So to have a TOC that shows all the index would still be better. 

Example:

https://airflow.readthedocs.io/en/stable/
image

https://airflow.readthedocs.io/en/stable/howto/index.html
image

https://airflow.readthedocs.io/en/stable/howto/operator/gcp/index.html
image

**Comment by mik-laj** [comment]
What do you think about moving the entire content to a separate sub-page with the title "Overview" and leaving the only a table of contents in the index.rst file?
Demo:
https://www.sphinx-doc.org/en/master/contents.html
https://sphinx-autoapi.readthedocs.io/en/latest/
The index.htmk website is very easy to access as it is not available in the TOC

**Comment by ashb** [comment]
Whops, just saw your comment as I hit merge @mik-laj, sorry!

**Comment by ashb** [comment]
I think index.html shouldn't be the index page, but some kind of welcome/intro -- it's often the first place people land, and going straight to "here's a load of links" isn't very helpful.

I have no problem with a separate contents page.

**Comment by mik-laj** [comment]
What do you think about having index.html to have a table of contents in TOC, but to give a liink to overview.html on s.apache.org/aiirflow-docs ?
The link to index.html is not available in the side menu and users are having trouble finding this page.

Do you know how to go to index.html while you are at tutorial.html?
http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/latest/tutorial.html

**Comment by ashb** [comment]
I don't on our theme, but in the default RTD theme it's here:

image

**Comment by ashb** [comment]
Testing a fix for this anyway @mik-laj

**Pull request merged**